files/external_node_v2.rb

```diff
@@ -163,6 +165,8 @@ def generate_fact_request(certname, filename)
     rescue => e
       raise "Could not generate facts for Foreman: #{e}"
     end
+  else
+    raise 'Last facts are older than last upload time, rerun puppet agent on host to get newer facts'
```

Comment:
there's a raise few lines above so I took the same approach, I'll change to `$stderr.puts`